assets/app/features/upgrade/upgrade.controller.spec.js

```diff
@@ -44,14 +54,51 @@ describe('Upgrade Controller', function () {
             should.exist(controller.steps);
         });
 
-        it('should contains a nextStep function ', function() {
-            should.exist(controller.steps.nextStep);
-            expect(controller.steps.nextStep).toEqual(jasmine.any(Function));
+        describe('nextStep function', function () {
+            it('should exist', function () {
+                should.exist(controller.steps.nextStep);
+                expect(controller.steps.nextStep).toEqual(jasmine.any(Function));
+            });
+
+            it('when called it should move to the next step', function () {
+                expect($state.current.id).toBe(0);
+                // calculate the next step in the list
+                var nextStep = (controller.steps.list[upgradeStepsFactory.activeStep.id + 1]).state;
+                controller.steps.nextStep();
+                expect($state.go).toHaveBeenCalledWith(nextStep);
+            });
+
+            it('when called in the last step it should not change steps', function () {
+                // set the last step in the list as current
+                $state.current = {id: 0, name: 'upgrade.upgrade-nodes', state: 'upgrade.upgrade-nodes'};
+                // refresh the list so it sets it as active
+                upgradeStepsFactory.refeshStepsList();
```

Comment:
`upgradeStepsFactory` is an external component that should be tested independently. In order to unit test the upgradeController, this factory needs to be mocked/injected and manipulated as part of the preconditions.

So instead of `upgradeStepsFactory.refeshStepsList();`, I think the following could be done:
```
upgradeStepsFactory.activeStep = upgradeStepsFactory.steps.list[upgradeStepsFactory.steps.list.length - 1];
```